python/tk_multi_launchapp/software_entity_launcher.py

```diff
@@ -26,74 +65,85 @@ def register_launch_commands(self):
+                    sw_entity.get("sg_mac_path") is None and \
+                    sw_entity.get("sg_linux_path") is None:
+
+                # all paths are none - we are in automatic mode
+                self._tk_app.log_debug("All path fields are None. Automatic mode.")
+
+                # defer to the automatic DCC scan to enumerate and register DCCs
+                self._scan_for_software_and_register(
+                    engine_str,
+                    dcc_versions,
+                    app_group,
+                    is_group_default
                 )
-                if not app_engine_path:
-                    self._tk_app.log_warning(
-                        "Software engine %s is not loaded in the current environment. "
-                        "Setting Software %s 'engine' to None." % (app_engine, app_display_name)
+
+            else:
+                # one or more path fields are not none. This means manual mode.
+                self._tk_app.log_debug("One or more path fields are not None. Manual mode.")
+
+                # Resolve the app path and args field names for the current platform
+                app_path_field = "sg_%s_path" % self._platform_name
+                app_args_field = "sg_%s_args" % self._platform_name
+
+                if sw_entity[app_path_field] is None:
+                    # manual mode but nothing to do for our os
+                    self._tk_app.log_debug(
+                        "No path defined for current platform (field %s) - skipping." % app_path_field
                     )
-                    app_engine = None
-
-            # Get the list of command data dictionaries from the information provided by this
-            # Software entity
-            register_cmd_data.extend(self._build_register_command_data(
-                app_display_name, app_icon, app_engine, app_path, app_args, app_versions,
-                app_group, is_group_default, sw_entity["type"], sw_entity["id"]
-            ))
-
-        # Use the BaseLauncher._register_launch_command() to register command
-        # data with the current engine.
-        for register_cmd in register_cmd_data:
-            self._register_launch_command(
-                register_cmd["display_name"],
-                register_cmd["icon"],
-                register_cmd["engine"],
-                register_cmd["path"],
-                register_cmd["args"],
-                register_cmd["version"],
-                register_cmd["group"],
-                register_cmd["group_default"],
-            )
+                    continue
+
+                app_path = sw_entity[app_path_field]
+                app_display_name = sw_entity["code"]
+                app_args = sw_entity[app_args_field] or ""
+
+                # get icon
+                icon_path = self._extract_thumbnail(
+                    sw_entity["type"],
+                    sw_entity["id"],
+                    sw_entity["image"]
+                )
```

Comment:
As discussed in our call, let's keep things simple for now - when in "automatic mode", the icon is picked up from the software scanning and when in "manual mode", it's picked up from Shotgun.